aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -457,6 +463,153 @@ std::tuple<Tensor&, Tensor&> min_out(
   return result;
 }
 
+Tensor& clamp_out(const Tensor& self, const c10::optional<Scalar>& min, const c10::optional<Scalar>& max, Tensor& result) {
+  if (min && max) {
+    auto iter = TensorIterator::unary_op(result, self);
+    clamp_scalar_stub(iter.device_type(), iter, *min, *max);
+  } else if (max) {
+    at::clamp_max_outf(self, *max, result);
+  } else if (min) {
+    at::clamp_min_outf(self, *min, result);
+  } else {
+    TORCH_CHECK(false, "torch.clamp: At least one of 'min' or 'max' must not be None");
+  }
+  return result;
+}
+
+Tensor& clamp_out(const Tensor& self, const c10::optional<Tensor>& min,
+                  const c10::optional<Tensor>& max, Tensor& result) {
+  if (min && max) {
+    TORCH_CHECK(self.layout() == Layout::Strided,
+                "torch.clamp only supports strided layout, got: ", self.layout());
+    auto iter = TensorIteratorConfig()
+                .set_check_mem_overlap(true)
+                .add_output(result)
+                .add_input(self)
+                .add_input(*min)
+                .add_input(*max)
+                .promote_inputs_to_common_dtype(true)
+                .cast_common_dtype_to_outputs(true)
+                .enforce_safe_casting_to_output(true)
+                .build();
+    clamp_stub(iter.device_type(), iter);
+  } else if (max) {
+    at::clamp_max_outf(self, *max, result);
+  } else if (min) {
+    at::clamp_min_outf(self, *min, result);
+  } else {
+    TORCH_CHECK(false, "torch.clamp: At least one of 'min' or 'max' must not be None");
+  }
+  return result;
+}
+
+Tensor clamp(const Tensor& self, const c10::optional<Scalar>& min, const c10::optional<Scalar>& max) {
```

Comment:
The scalar versions of clamp aren't changing with this PR, right? These modifications (and the test modifications in common_methods_invocations.py) are just because the code is moving around?

Reply:
Yeah, it just made more sense to me that they all be in the same file.

Reply:
Agreed.